branching_xblock/branching_xblock.py

```diff
@@ -93,30 +98,33 @@ def start_node(self):
         """
         if not self.current_node_id and self.scenario_data["start_node_id"]:
             self.current_node_id = self.scenario_data["start_node_id"]
-    
+
     def get_node(self, node_id):
         """
         Get a node by its ID
         """
+        # Since the order of nodes doesn't matter, why not make it a dictionary so you can
+        # get the node as self.scenario_data["nodes"][node_id]
         return next(
             (node for node in self.scenario_data["nodes"] if node["id"] == node_id),
             None
         )
-    
+
     def get_current_node(self):
         """
         Get the learner's current node
         """
         return self.get_node(self.current_node_id) if self.current_node_id else None
-    
+
     def is_start_node(self, node_id):
+        # Couldn't we also get this from 'start_node_id' ?
```

Comment:
Yeah this method was left over from one of the iterations of the solution - sorry about that. Will clean it up.

Reply:
Done.